src/util/hash.rs

```diff
@@ -64,7 +64,7 @@ pub fn bitcoin_merkle_root<T, I>(mut iter: I) -> T
         return Default::default();
     }
     if iter.len() == 1 {
-        return T::from_inner(iter.next().unwrap().into_inner());
+        return iter.next().unwrap();
     }
```

Comment:
I really dislike unwraps in the code. May we have
```suggestion
if let Some(item) = iter.next() {
     return item
}
```

Reply:
Most certainly - death to `unwrap`.

Reply:
`if let Some(item) ...` after already checking the length looks a bit odd. I elected to use an `expect` instead. For comparison here they both are. I've pushed the `expect` version but can change it if folk prefer the `if let Some` version.
```
    if iter.len() == 1 {
        if let Some(hash) = iter.next() {
            return hash;
        }
    }
```
vs
```
    if iter.len() == 1 {
        return iter.next().expect("cannot fail, length == 1");
    }
```

Reply:
Yeah, I would've suggested it too but the existing check also checks it's not more than 1, so I'm not sure if it can be made better.

Reply:
@tcharding yeah, it seems this can't be made better avoiding `expect`